commands/index.go

```diff
@@ -0,0 +1,202 @@
+	lotuscli "github.com/filecoin-project/lotus/cli"
+	"github.com/ipfs/go-cid"
+	"github.com/urfave/cli/v2"
+	"golang.org/x/xerrors"
+
+	"github.com/filecoin-project/lily/chain/actors/builtin"
+	"github.com/filecoin-project/lily/chain/indexer"
+	"github.com/filecoin-project/lily/lens/lily"
+)
+
+type indexOps struct {
+	tasks    string
+	storage  string
+	apiAddr  string
+	apiToken string
+	name     string
+	window   time.Duration
+}
+
+var indexFlags indexOps
+
+var IndexTipSetCmd = &cli.Command{
+	Name:  "tipset",
+	Usage: "Index the state of a tipset from the filecoin blockchain by tipset key",
+	Action: func(cctx *cli.Context) error {
+		ctx := lotuscli.ReqContext(cctx)
+
+		indexName := fmt.Sprintf("index_%d", time.Now().Unix())
+		if indexFlags.name != "" {
+			indexName = indexFlags.name
+		}
+
+		var tsStr string
+		if tsStr = cctx.Args().First(); tsStr == "" {
+			return xerrors.Errorf("tipset argument required")
+		}
+
+		tsk, err := parseTipSetKey(strings.Split(tsStr, ","))
+		if err != nil {
+			return xerrors.Errorf("failed to parse tipset key: %w", err)
+		}
+
+		tasks := strings.Split(indexFlags.tasks, ",")
+		if indexFlags.tasks == "*" {
+			tasks = indexer.AllTasks
+		}
+
+		cfg := &lily.LilyIndexConfig{
+			TipSet:  tsk,
+			Name:    indexName,
+			Tasks:   tasks,
+			Storage: indexFlags.storage,
+			Window:  indexFlags.window,
+		}
+
+		api, closer, err := GetAPI(ctx, indexFlags.apiAddr, indexFlags.apiToken)
+		if err != nil {
+			return err
+		}
+		defer closer()
+
+		_, err = api.LilyIndex(ctx, cfg)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	},
+}
+
+var IndexHeightCmd = &cli.Command{
+	Name:  "height",
+	Usage: "Index the state of a tipset from the filecoin blockchain by height",
+	Action: func(cctx *cli.Context) error {
+		ctx := lotuscli.ReqContext(cctx)
+
+		indexName := fmt.Sprintf("index_%d", time.Now().Unix())
+		if indexFlags.name != "" {
+			indexName = indexFlags.name
+		}
+
+		var tsStr string
+		if tsStr = cctx.Args().First(); tsStr == "" {
+			return xerrors.Errorf("height argument required")
+		}
+
+		api, closer, err := GetAPI(ctx, indexFlags.apiAddr, indexFlags.apiToken)
+		if err != nil {
+			return err
+		}
+		defer closer()
+
+		height, err := strconv.ParseInt(cctx.Args().First(), 10, 46)
+		if err != nil {
+			return err
+		}
+		ts, err := api.ChainGetTipSetByHeight(ctx, abi.ChainEpoch(height), types.EmptyTSK)
+		if err != nil {
+			return err
+		}
+
+		if height != int64(ts.Height()) {
+			log.Warnf("height (%d) is null round, indexing height %d", height, ts.Height())
+		}
+
+		tasks := strings.Split(indexFlags.tasks, ",")
+		if indexFlags.tasks == "*" {
+			tasks = indexer.AllTasks
+		}
+
+		cfg := &lily.LilyIndexConfig{
+			TipSet:  ts.Key(),
+			Name:    indexName,
+			Tasks:   tasks,
+			Storage: indexFlags.storage,
+			Window:  indexFlags.window,
+		}
+
+		_, err = api.LilyIndex(ctx, cfg)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	},
+}
+
+var IndexCmd = &cli.Command{
```

Comment:
New command can be called like:
```
lily index --storage=Database1 --name=testing --window=5m --tasks="blocks" height 1578553
```
or
```
lily index --storage=Database1 --name=testing --window=5m --tasks=blocks tipset bafy2bzacecaoicpyxikq3yvabri5yvscc42hvvikrtdk73kiavog45wi4gify
```

I find this easier than calling walk with the same from and too heights.